cli/src/template_builder.rs

```diff
@@ -605,6 +605,39 @@ fn build_binary_operation<'a, L: TemplateLanguage<'a> + ?Sized>(
             let out = lhs.and_then(move |l| Ok(l && rhs.extract()?));
             Ok(L::wrap_boolean(out))
         }
+        BinaryOp::LogicalEq => {
+            let lhs = build_expression(language, diagnostics, build_ctx, lhs_node)?;
+            let rhs = build_expression(language, diagnostics, build_ctx, rhs_node)?;
+            match (lhs.type_name(), rhs.type_name()) {
+                ("Boolean", "Boolean") => {
+                    let lhs = lhs.try_into_boolean().unwrap();
+                    let rhs = rhs.try_into_boolean().unwrap();
+                    let out = lhs.and_then(move |l| Ok(l == rhs.extract()?));
+                    Ok(L::wrap_boolean(out))
+                }
+                ("Integer", "Integer") => {
+                    let lhs = lhs.try_into_integer().unwrap();
+                    let rhs = rhs.try_into_integer().unwrap();
+                    let out = lhs.and_then(move |l| Ok(l == rhs.extract()?));
+                    Ok(L::wrap_boolean(out))
+                }
+                ("String", "String") => {
+                    let lhs = lhs.try_into_plain_text().unwrap();
+                    let rhs = rhs.try_into_plain_text().unwrap();
+                    let out = lhs.and_then(move |l| Ok(l == rhs.extract()?));
+                    Ok(L::wrap_boolean(out))
+                }
+                (lhs_type @ ("Boolean" | "Integer" | "String"), rhs_type) => Err(
+                    TemplateParseError::expected_type(lhs_type, rhs_type, rhs_node.span),
+                ),
+                (lhs_type, _) => {
+                    let message = format!(
+                        r#"Expected expression of type "Boolean", "Integer", or "String", but actual type is "{lhs_type}""#
+                    );
+                    Err(TemplateParseError::expression(message, lhs_node.span))
+                }
```

Comment:
nit: Perhaps, this will be a method of `Property` type (something like `lhs.try_eq(rhs) -> Option<Box<dyn TemplateProperty<Output = bool>>>`), but it can be implemented later.

Can you leave a TODO comment?